lib/clients/analytics.ts

```diff
@@ -0,0 +1,131 @@
+import Analytics from '@rudderstack/rudder-sdk-node';
+import { GeneralTrace } from '@voiceflow/general-types';
+import { AxiosResponse } from 'axios';
+
+import log from '@/logger';
+import { Config } from '@/types';
+
+import IngestApi, { EventsType, InteractBody } from './ingest-client';
+
+export class AnalyticsSystem {
+  private analyticsClient: any;
+
+  private aggregateAnalytics = false;
+
+  private ingestClient?: IngestApi;
+
+  constructor(config?: Config) {
+    if (config) {
+      if (config.ANALYTICS_WRITE_KEY && config.ANALYTICS_ENDPOINT) {
+        this.analyticsClient = new Analytics(config.ANALYTICS_WRITE_KEY, `${config.ANALYTICS_ENDPOINT}/v1/batch`);
+      }
+
+      if (config.INGEST_WEBHOOK_ENDPOINT) {
+        this.ingestClient = new IngestApi(config.INGEST_WEBHOOK_ENDPOINT, undefined);
+      }
+      this.aggregateAnalytics = true;
+    }
+  }
+
+  identify(id: string) {
+    const payload = {
+      userId: id,
+    };
+    if (this.aggregateAnalytics && this.analyticsClient) {
+      log.trace('Identify');
+      this.analyticsClient.identify(payload);
+    }
+  }
+
+  private callAnalyticsSystemTrack(id: string, eventId: string, metadata: any) {
+    const interactAnalyticsBody = {
+      userId: id,
+      event: eventId,
+      properties: {
+        metadata,
+      },
+    };
+    this.analyticsClient.track(interactAnalyticsBody);
+  }
+
+  private createInteractBody(id: string, eventId: string, metadata: any): InteractBody {
+    let sessionId: string;
+    if (metadata.data && metadata.data.reqHeaders && metadata.data.reqHeaders.sessionid) {
```

Comment:
We can use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to clean up logic like this:
```suggestion
    if (metadata.data?.reqHeaders?.sessionid) {
```

Reply:
fixed!